src/tir/pass/unroll_loop.cc

```diff
@@ -160,7 +160,9 @@ class LoopUnroller : public StmtExprMutator {
     PrimExpr extent = tir::Simplify(op->extent);
     const IntImmNode  *v1 = extent.as<IntImmNode>();
     int value = -1;
-    if (v1 != nullptr) {
+    // integers that do not fit in int32_t are treated as symbolic,
+    // as it's impossible to unroll such large loops
+    if (v1 != nullptr && v1->value <= std::numeric_limits<int>::max()) {
```

Comment:
I wonder if we should use `int32_t` here rather than `int`. I'm not sure, but just worry about `int` will represent different types (`int16_t`, `int32_t` or `int64_t`) on different systems and devices.

Reply:
My motivation here is to prevent overflow in the next line (which uses int):

```
value = static_cast<int>(v1->value);
```

IMO it might be fine to use `int` here, as it's consistent with other parts of the pass. What do you think?

Reply:
That's fine